File: mobile-android/app/src/main/java/com/legalsaas/android/data/model/Case.kt
package com.legalsaas.android.data.model

import com.google.gson.annotations.SerializedName

data class Case(
    @SerializedName("id")
    val id: String,

    @SerializedName("title")
    val title: String,

    @SerializedName("description")
    val description: String? = null,

    @SerializedName("case_number")
    val caseNumber: String,

    @SerializedName("status")
    val status: CaseStatus,

    @SerializedName("priority")
    val priority: CasePriority,

    @SerializedName("client_name")
    val clientName: String,

    @SerializedName("client_email")
    val clientEmail: String? = null,

    @SerializedName("client_phone")
    val clientPhone: String? = null,

    @SerializedName("court_name")
    val courtName: String? = null,

    @SerializedName("judge_name")
    val judgeName: String? = null,

    @SerializedName("opposing_counsel")
    val opposingCounsel: String? = null,

    @SerializedName("practice_area")
    val practiceArea: String? = null,

    @SerializedName("filing_date")
    val filingDate: String? = null,

    @SerializedName("next_hearing_date")
    val nextHearingDate: String? = null,

    @SerializedName("assigned_attorney_id")
    val assignedAttorneyId: String? = null,

    @SerializedName("created_at")
    val createdAt: String,

    @SerializedName("updated_at")
    val updatedAt: String
)

enum class CaseStatus {
    @SerializedName("open")
    OPEN,

    @SerializedName("in_progress")
    IN_PROGRESS,

    @SerializedName("pending")
    PENDING,

    @SerializedName("closed")
    CLOSED,

    @SerializedName("dismissed")
    DISMISSED,

    @SerializedName("settled")
    SETTLED
}

enum class CasePriority {
    @SerializedName("low")
    LOW,

    @SerializedName("medium")
    MEDIUM,

    @SerializedName("high")
    HIGH,

    @SerializedName("urgent")
    URGENT
}

data class CaseRequest(
    @SerializedName("title")
    val title: String,

    @SerializedName("description")
    val description: String? = null,

    @SerializedName("case_number")
    val caseNumber: String,

    @SerializedName("status")
    val status: CaseStatus,

    @SerializedName("priority")
    val priority: CasePriority,

    @SerializedName("client_name")
    val clientName: String,

    @SerializedName("client_email")
    val clientEmail: String? = null,

    @SerializedName("client_phone")
    val clientPhone: String? = null,

    @SerializedName("court_name")
    val courtName: String? = null,

    @SerializedName("judge_name")
    val judgeName: String? = null,

    @SerializedName("opposing_counsel")
    val opposingCounsel: String? = null,

    @SerializedName("practice_area")
    val practiceArea: String? = null,

    @SerializedName("filing_date")
    val filingDate: String? = null,

    @SerializedName("next_hearing_date")
    val nextHearingDate: String? = null,

    @SerializedName("assigned_attorney_id")
    val assignedAttorneyId: String? = null
)

File: mobile-ios/LegalSaaS/Models/Case.swift
import Foundation

struct Case: Codable, Identifiable {
    let id: UUID
    let title: String
    let description: String?
    let caseNumber: String
    let status: CaseStatus
    let priority: CasePriority
    let clientName: String
    let clientEmail: String?
    let clientPhone: String?
    let courtName: String?
    let judgeName: String?
    let assignedTo: UUID
    let createdAt: Date
    let updatedAt: Date
    let deadlines: [Deadline]?

    enum CodingKeys: String, CodingKey {
        case id, title, description, status, priority
        case caseNumber = "case_number"
        case clientName = "client_name"
        case clientEmail = "client_email"
        case clientPhone = "client_phone"
        case courtName = "court_name"
        case judgeName = "judge_name"
        case assignedTo = "assigned_to"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case deadlines
    }
}

enum CaseStatus: String, Codable, CaseIterable {
    case active
    case pending
    case closed
    case archived

    var displayName: String {
        switch self {
        case .active: return "Active"
        case .pending: return "Pending"
        case .closed: return "Closed"
        case .archived: return "Archived"
        }
    }

    var color: String {
        switch self {
        case .active: return "green"
        case .pending: return "orange"
        case .closed: return "gray"
        case .archived: return "blue"
        }
    }
}

enum CasePriority: String, Codable, CaseIterable {
    case low
    case medium
    case high
    case urgent

    var displayName: String {
        switch self {
        case .low: return "Low"
        case .medium: return "Medium"
        case .high: return "High"
        case .urgent: return "Urgent"
        }
    }
}

struct CreateCaseRequest: Codable {
    let title: String
    let description: String?
    let caseNumber: String
    let clientName: String
    let clientEmail: String?
    let clientPhone: String?
    let courtName: String?
    let judgeName: String?
    let priority: CasePriority

    enum CodingKeys: String, CodingKey {
        case title, description, priority
        case caseNumber = "case_number"
        case clientName = "client_name"
        case clientEmail = "client_email"
        case clientPhone = "client_phone"
        case courtName = "court_name"
        case judgeName = "judge_name"
    }
}
